scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+Contains functions for the project manager to call that gather data from o3de scripts and massage it
+"""
+
+import logging
+import pathlib
+
+from o3de import cmake, disable_gem, download, enable_gem, engine_properties, engine_template, manifest, project_properties, register, repo
+
+logger = logging.getLogger('o3de.project_manager_interface')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+
+def get_engine_info(engine_name: str = None, engine_path: str or pathlib.Path = None) -> dict or None:
```

Comment:
We should probably pick `str` or `pathlib.Path`, my thinking is use `str` and converting to `Path` in Python for simplicity.  Then we could remove the current c++ code that converts to `Path` basically doing the same thing with pybind here https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/PythonBindings.cpp#L60

Reply:
Good idea, changed to only using str!